simplipy/api.py

```diff
@@ -270,14 +295,17 @@ def is_fatal_error(err: ClientResponseError) -> bool:
                 which is where this error can occur; we can't control when/how that
                 happens (e.g., we might query the API in the middle of a base station
                 update), so it should be viewed as retryable.
+        3. 404: When fetching media files, you may get a 404 if the media file is not
```

Comment:
This is only one scenario in which a 404 should be viewed as worth a retry; in all others, it's a legitimate error that shouldn't be retried. I don't think we should do this for all requests.

Reply:
I understand. I'll fix it.

Reply:
Made it reusable by passing in retry codes.